tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml_iteration.bal

```diff
@@ -308,6 +308,40 @@ function testSequenceOfSequenceOfXmlElementIteration() {
     validateValues(arr);
 }
 
+function testSequenceOfSequenceOfReadonlyXmlElementIteration() {
+    xml<xml<xml:Element & readonly>> elements = xml `<foo/><bar>value</bar><baz>1</baz>`;
+    xml:Element[] arr = [];
+    foreach var e1 in elements {
+        arr.push(e1);
+    }
+    validateValues(arr);
+
+    xml<xml:Element & readonly> elements2 = xml `<foo/><bar>value</bar><baz>1</baz>`;
+    arr = [];
+    foreach var e2 in elements2 {
+        arr.push(e2);
+    }
+    validateValues(arr);
+}
+
+function testSequenceOfXmlReadonlyUnionIteration() {
```

Comment:
Maybe
```suggestion
function testSequenceOfReadOnlyXmlSubTypeUnionIteration() {
```

Kinda reads like a union of xml and readonly atm.

Reply:
Updated - 162c9db61a2443f5e02a97993461b5f6b88bc530